src/devices/bus/coco/coco_fdc.cpp

```diff
@@ -2,115 +2,121 @@
+
+	DSKREG - the control register
+	CoCo ($FF40)
+
+	Bit
+	7 halt enable flag
+	6 drive select #3
+	5 density (0=single, 1=double) and NMI enable flag
+	4 write precompensation
+	3 drive motor activation
+	2 drive select #2
+	1 drive select #1
+	0 drive select #0
+
+	Reading from $FF48-$FF4F clears bit 7 of DSKREG ($FF40)
+
+	---------------------------------------------------------------------------
+
+	Disto Super Controller II
+		Mini Expansion Bus
+		No Halt Extension. Implemented by using a read and write cache.
+
+			CachDat - Cache Data Register
+				$FF74 & $FF75: Read/Write cache data.
+
+				Normally cache size is 256 bytes. But a common upgrade is to change
+				it to 512 bytes.
+
+			CachCtrl - Cache Controller
+				$FF76 & $FF77:
+					write
+						clears buffer counter
+
+					read bit:
+						7:  0 - indicates an interrupt request from the disk controller
+							1 - no interrupt
+						3:  0 - disable CART / FIRQ (masked)
+							1 - enable CART / FIRQ
+						2:  0 - normal NMI
+							1 - masked NMI
+						1:  0 - compatible I/O mode
+							1 - buffered I/O mode
+						0:  0 - buffered write operation
+							1 - buffered read operation
+
+		There was a jumper to set the addresses to be FF58 to FF5B. Not implemented.
 
 *********************************************************************/
 
 #include "emu.h"
-#include "cococart.h"
 #include "coco_fdc.h"
-#include "imagedev/floppy.h"
-#include "machine/msm6242.h"
-#include "machine/ds1315.h"
-#include "machine/wd_fdc.h"
-#include "machine/ram.h"
+#include "cococart.h"
 #include "formats/dmk_dsk.h"
+#include "formats/fs_coco_os9.h"
+#include "formats/fs_coco_rsdos.h"
 #include "formats/jvc_dsk.h"
-#include "formats/vdk_dsk.h"
-#include "formats/sdf_dsk.h"
 #include "formats/os9_dsk.h"
-#include "formats/fs_coco_rsdos.h"
-#include "formats/fs_coco_os9.h"
+#include "formats/sdf_dsk.h"
+#include "formats/vdk_dsk.h"
+#include "imagedev/floppy.h"
+#include "machine/ds1315.h"
+#include "machine/input_merger.h"
+#include "machine/msm6242.h"
+#include "machine/wd_fdc.h"
+#include "meb_intrf.h"
 
 
-//#define LOG_GENERAL   (1U << 0) //defined in logmacro.h already
-#define LOG_WDFDC   (1U << 1) // Shows register setup
-#define LOG_WDIO    (1U << 2) // Shows Data read and write
+//#define LOG_GENERAL	(1U << 0) //defined in logmacro.h already
+#define LOG_WDFDC	(1U << 1) // Shows register setup
+#define LOG_WDIO	(1U << 2) // Shows data read and write
+#define LOG_WDSCII	(1U << 3) // Shows SCII register setup
 
-//#define VERBOSE (LOG_GENERAL | LOG_WDFDC)
+//#define VERBOSE (LOG_GENERAL|LOG_WDFDC|LOG_WDSCII)
 #include "logmacro.h"
 
-#define LOGWDFDC(...)   LOGMASKED(LOG_WDFDC,  __VA_ARGS__)
-#define LOGWDIO(...)   LOGMASKED(LOG_WDIO,  __VA_ARGS__)
+#define LOGWDFDC(...)	LOGMASKED(LOG_WDFDC,  __VA_ARGS__)
+#define LOGWDIO(...)   LOGMASKED(LOG_WDIO,	__VA_ARGS__)
+#define LOGWDSCII(...)	LOGMASKED(LOG_WDSCII,  __VA_ARGS__)
```

Comment:
If you’re going to tabulate stuff, do it properly, otherwise use one space.  This is all over the place.

Reply:
Fixed.